src/FixedPointNumbers.jl

```diff
@@ -40,54 +41,87 @@ export
     # Functions
     scaledual
 
-reinterpret(x::FixedPoint) = x.i
+getindex(x::AbstractFixedPoint) = x.i
```

Comment:
I see your point, I will change it to raw.

On Tue, 3 Nov 2015 18:33 Tim Holy <notifications@github.com> wrote:

> In src/FixedPointNumbers.jl
> <https://github.com/JeffBezanson/FixedPointNumbers.jl/pull/35#discussion_r43728468>
> :
>
> > @@ -40,54 +41,87 @@ export
> >      # Functions
> >      scaledual
> >
> > -reinterpret(x::FixedPoint) = x.i
> > +getindex(x::AbstractFixedPoint) = x.i
>
> The problem, however, with your solution is that it conflicts with
>
> julia> 5[]5
>
> julia> 3.2[]3.2
>
> So anyone who writes code like
>
> function mysum(X)
>     s = zero(first(X))
>     for x in X
>         s += x
>     end
>     s
>
> will get a very surprising answer from mysum(0xffuf8).
>
> x.i might be used inside FixedPointNumbers, but I've never seen such
> usage outside. If this is intended only for internal use, rather than
> overload an exported function from Base you could define raw (or
> something) as an internal function.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/JeffBezanson/FixedPointNumbers.jl/pull/35/files#r43728468>
> .
>